src/Quartz/Core/QuartzScheduler.cs

```diff
@@ -1265,6 +1266,7 @@ public virtual Task<IJobDetail> GetJobDetail(JobKey jobKey)
 
             return await resources.JobStore.RetrieveTrigger(triggerKey).ConfigureAwait(false);
```

Comment:
It is necessary because of how `Task<T>` is designed. It doesn't support the necessary variance to be able to treat `Task<IOperableSomething>` as the defined returned type of the method.